tests/e2e/test_file_upload.py

```
@@ -1,86 +1,222 @@
+        auth_file_input = authenticated_page.query_selector(auth_home.FILE_UPLOAD)
+        auth_upload_available = auth_file_input and auth_file_input.is_visible() and not auth_file_input.is_disabled()
+        
+        # Test unauthenticated user
+        unauth_home = HomePage(page_with_base_url)
+        unauth_home.navigate_to_home()
+        
+        unauth_file_input = page_with_base_url.query_selector(unauth_home.FILE_UPLOAD)
+        unauth_upload_available = unauth_file_input and unauth_file_input.is_visible() and not unauth_file_input.is_disabled()
+        
+        # Document the difference
+        assert True, f"Upload availability - Authenticated: {auth_upload_available}, Unauthenticated: {unauth_upload_available}"
```

Ruff (0.12.2)

Use of `assert` detected

(S101)